### ISSUE TITLE
Drag and Pan Feature

### ISSUE THREAD

**Issue by zhpeng811** [iGeek-X-Lab/CS_Course_Ladder#6]
Invest into the drag and pan feature for the GUI, need to make sure the buttons are still clickable after dragging or panning

Here is some information provided by Kingston
- I recommend wrapping the React chart in a <canvas> or <div> that is handled with a drag & pan capability: https://codepen.io/SpacecaseArtist/pen/GvybJp
- That way, all the contents within React is in DOM container of its own, and be managed by a click-n-drag mechanism from a higher level
- https://reactjs.org/docs/rendering-elements.html (showing how to target and render React content to a specific div on a page)
- In a nutshell, you can also treat the React course charts as a giant image
